### PR TITLE
Add better retry and kickoff instructions

### PR DESCRIPTION
**Pull request by dagood**
This PR rearranges the guidance on the main instruction page a bit to give more obvious step-by-step instructions and defer more of the retry guidance to the Extensions page.

The PR also adds a lot to the retry guidance, and uses text/template to generate it rather than doing it all with a stringbuilder and simple string literals. This makes it easier to author because you can use the markdown preview in (e.g.) VS Code. I also added golden-file-style tests so it's easier to see the final output and make sure it doesn't give unexpected results. This also helps make sure the template logic doesn't break in CI.

The build failure guidance doc is now a lot more specific and even in some cases includes direct links to the PR or AzDO build that you need to check rather than telling the dev to go looking through the raw build logs for URLs.

(I made a folder to store more templates, because I expect to use more in the future, e.g. for GitHub issue and PR content.)

Rendered main instruction page:
https://github.com/microsoft/go-infra/blob/dev/dagood/useful-extdoc/docs/release-process/instructions.md

Rendered example retry instructions:
https://github.com/microsoft/go-infra/tree/dev/dagood/useful-extdoc/cmd/releasego/testdata/retry-instructions

Example extensions page with the new guidance (includes a garbage PR number--just for visualization):
https://dev.azure.com/dnceng/internal/_build/results?buildId=2006597&view=ms.vss-build-web.run-extensions-tab

### CODE REVIEW

**Review by qmuntal — APPROVED** [review]
LGTM++. I've tried hard to find a typo or something to improve, but this PR is really neat.

File: docs/release-process/instructions.md [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
 > * This makes infrastructure safer: if the build gets triggered and the set of parameters isn't passed correctly by tooling, the build fails safe by refusing to start.
 
-1. Once the values are set, run the build.
-    * It will take some time to reserve a build agent. Then, it creates one issue to track the release day progress and one issue per version in the list.
+For example, a security patch release for 1.17 and 1.18 may look like this:
+
+> ![](images/run-release-start.png)
+
+1. Now that the values are set, press Run.
+    * It will take some time to reserve a build agent. Expect up to ten minutes. Then, the job creates one issue to track the release day progress and one issue per version in the list.
```
Comment (jaredpar) [inline docs/release-process/instructions.md:34]: Should mention where to look for the issue link.

File: cmd/releasego/add-retry-instructions.go [hunk 2]
```diff
@@ -131,3 +100,69 @@ func handleAddRetryInstructions(p subcmd.ParseFunc) error {
 
 	return nil
 }
+
+// Match env vars like "poll1buildId=123" with the number and value in groups. If there are
+// multiple "=", treat the leftmost one as the separator. Ignore case.
+var reg = regexp.MustCompile(`(?i)poll(\d+)(.+?)=(.*)`)
```
Comment (jaredpar) [inline cmd/releasego/add-retry-instructions.go:106]: What is `(?`?

File: cmd/releasego/add-retry-instructions.go [hunk 3]
```diff
@@ -131,3 +100,69 @@ func handleAddRetryInstructions(p subcmd.ParseFunc) error {
...
+	Index int
+}
+
+func newEnvArg(envVar string) (*envArg, error) {
+	matches := reg.FindStringSubmatch(envVar)
+	if len(matches) > 0 {
+		fmt.Printf("Found polling env var %q. Matches: %#v\n", envVar, matches)
+		num, err := strconv.Atoi(matches[1])
+		if err != nil {
+			return nil, fmt.Errorf("match 1 is not an int: %w", err)
```
Comment (jaredpar) [inline cmd/releasego/add-retry-instructions.go:120]: Consider putting `matches[1]` in the error string so we know what didn't parse.

Comment (dagood) [reply cmd/releasego/add-retry-instructions.go:120]: The value actually shows up in `err` already, e.g. `strconv.Atoi: parsing "not-a-number": invalid syntax`, `strconv.Atoi: parsing "79832459834523534534532425246246246246593847": value out of range`

File: cmd/releasego/add-retry-instructions.go (hunk 2 above)
Comment (dagood) [reply cmd/releasego/add-retry-instructions.go:106]: That sets flags: https://github.com/google/re2/wiki/Syntax `(?flags)` in "Grouping" table, `i` being case-insensitive.

**Review by jaredpar — APPROVED** [review]

**Pull request merged**